planner/optimize.go

```diff
@@ -273,13 +274,26 @@ func optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 	return finalPlan, names, cost, err
 }
 
-func extractSelectAndNormalizeDigest(stmtNode ast.StmtNode) (ast.StmtNode, string, string) {
+func extractSelectAndNormalizeDigest(stmtNode ast.StmtNode, defaultDB string, restore bool) (ast.StmtNode, string, string) {
```

Comment:
How about changing its arguments to `(stmtNode ast.StmtNode, specifiedDB string)` and using `specifiledDB` only when it is not empty?